### ISSUE TITLE
Allow users to fetch dbt projects from separate git repos

### ISSUE THREAD

**Issue by tatiana** [astronomer/astronomer-cosmos#518]
The community has reported more than once the need to have dbt projects and Airflow workflows in separate Git repos, such as:
https://apache-airflow.slack.com/archives/C059CC42E9W/p1694089200140099?thread_ts=1688876790.768469&cid=C059CC42E9W

This ticket aims to allow users to reference manifest/dbt models who live in a separate repo.

**Comment by tatiana** [comment]
We still need to shape this story - @jlaneve, do you have any suggestions?

**Comment by qimumu9406** [comment]
@tatiana I am looking forward to this function very much. In fact, airflow code and dbt code are different types of code. They should not be stored in the same directory or code library. If this function is implemented, it will be very friendly to cosmos (and dbt) on kubernetes. Usually dbt code Run in kubernetes pods as image

**Comment by joppedh** [comment]
This would be great. Keeping the separation of concerns between engineering and analytics

**Comment by peterampazzo** [comment]
To maintain the separation of the dagbag and dbt model, we're aiming to implement a pattern that involves utilizing [git-sync](https://github.com/kubernetes/git-sync) the source code of the dbt model into a PVC, which is then mounted by the Airflow scheduler.

**Comment by tatiana** [comment]
@qimumu9406 @joppedh @peterampazzo I just tried to add more details to how Cosmos GitHub support could look from a user perspective: #570

Since #570 focuses only on the DAG parsing side of things & Kubernetes/Docker execution mode, I didn't change the description of the current ticket (#518).

 @peterampazzo is very pragmatic and will work with Cosmos 1.0/1.1. In the past, I used similar solutions and they can work very well.

We can still think about alternative solutions that would work for `ExecutionMode.LOCAL`, which would not depend on infrastructure changes. 

One possibility could be to sync the dbt project pipelines code at the beginning of the Local Operator/task execution. WDYT?
